storage/database/db_manager.go

```diff
@@ -476,12 +476,76 @@ func (dbm *databaseManager) NewBatch(dbEntryType DBEntryType) Batch {
 		defer dbm.lockInMigration.RUnlock()
 
 		if dbm.inMigration {
-			return dbm.GetStateTrieMigrationDB().NewBatch()
+			newDBBatch := dbm.getDatabase(StateTrieMigrationDB).NewBatch()
+			oldDBBatch := dbm.getDatabase(StateTrieDB).NewBatch()
+			return NewStateTrieDBBatch([]Batch{oldDBBatch, newDBBatch})
 		}
+	} else if dbEntryType == StateTrieMigrationDB {
+		return dbm.GetStateTrieMigrationDB().NewBatch()
 	}
 	return dbm.getDatabase(dbEntryType).NewBatch()
 }
 
+func NewStateTrieDBBatch(batches []Batch) Batch {
+	return &stateTrieDBBatch{batches: batches}
+}
+
+type stateTrieDBBatch struct {
+	batches []Batch
+}
+
+func (stdBatch *stateTrieDBBatch) Put(key []byte, value []byte) error {
+	var errResult error
+	for _, batch := range stdBatch.batches {
+		if err := batch.Put(key, value); err != nil {
+			errResult = err
+		}
+	}
+
+	return errResult
+}
+
+// ValueSize is called to determine whether to write batches when it exceeds
+// certain limit. partitionedDB returns the largest size of its batches to
+// write all batches at once when one of batch exceeds the limit.
+func (stdBatch *stateTrieDBBatch) ValueSize() int {
+	maxSize := 0
+	for _, batch := range stdBatch.batches {
+		if batch.ValueSize() > maxSize {
+			maxSize = batch.ValueSize()
+		}
+	}
+
+	return maxSize
+}
+
+// Write passes the list of batch tasks to taskCh so batch can be processed
+// by underlying workers. Write waits until all workers return the result.
+func (stdBatch *stateTrieDBBatch) Write() error {
```

Comment:
There are functions such as `WriteBatches` and `WriteBatchesOverThreshold` and this seems it is duplicated code. What do you think about modifying the functions and calling the functions here?

Reply:
I think this stateTrieDBBatch is the lower level than `WriteBatches` and `WriteBatchesOverThreshold`.
They call state trie batch write and use `stateTrieDBBatch` in migration.
So, They are not duplicated.

Reply:
I don't get that you said it is **other level**. stdBatch can call `WriteBatches` like this without an inifinite loop : 
```
func (stdBatch *stateTrieDBBatch) Write() error {
	_, err := WriteBatches(stdBatch.batches...)
	return err
}
```
If you don't want to change the definition of `WriteBatches`, a new function can be defined :
```
func WriteBatchesParallel(batches ...Batch) (int, error) {
   ...
}

func (stdBatch *stateTrieDBBatch) Write() error {
	_, err := WriteBatchesParallel(stdBatch.batches...)
	return err
}
```
I still think the code can be reused in some place else and this code limits its usage in `stdBatch`.
What do you think? :)

Reply:
I reflected. Thanks. :) @winnie-byun PTAL.

Reply:
Thanks for considering. LGTM :)